### PR TITLE
remove core sched rotator checking. add whole queue kwarg to request_…

### PR DESCRIPTION
**Pull request by yoachim**
Remove some rotator checking that's in `CoreScheduler`.
Add a kwarg to return the whole observing queue rather than a single observation.

### CODE REVIEW

File: rubin_scheduler/scheduler/schedulers/core_scheduler.py [hunk 1]
```diff
@@ -213,21 +203,25 @@ def _check_queue_mjd_only(self, mjd):
...
             The Modified Julian Date.
             If None, it uses the MJD from the conditions from the
             last conditions update.
+        whole_queue : `bool`
+            Return the entire list of observations in the queue (True),
+            or just a single observation (False). Default False
 
         Returns
         -------
         observation :  observation object (ra,dec,filter,rotangle)
```
Comment (rhiannonlynne) [inline rubin_scheduler/scheduler/schedulers/core_scheduler.py:222]: Is this returning an ObservationArray now?

Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/schedulers/core_scheduler.py:222]: (or list of ObservationArray)?

**Comment by rhiannonlynne** [comment]
I'm guessing checking the rotator angle wasn't needed because the surveys didn't propose out of range rotation angles? Or is it just that the checking shouldn't happen here, and instead in the ModelObservatory/SchedulerCSC? 
What happens if a survey doesn't have a detailer to set rotation angles? (some of the auxtel surveys send '[]' and not None, which doesn't trigger the addition of the ZeroRotation detailer, but also I don't think that's quite what they want for the spectroscopic surveys ... but also i think those are probably having rotation angles added after the fact in some other CSC).

**Comment by rhiannonlynne** [comment]
This probably wants @tribeiro to comment.

File: rubin_scheduler/scheduler/schedulers/core_scheduler.py (hunk 1 above)
Comment (yoachim) [reply rubin_scheduler/scheduler/schedulers/core_scheduler.py:222]: List of ObservationArray.

Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/schedulers/core_scheduler.py:222]: Can you update this docstring please? 
probably something like 
observation : `list` [`~.scheduler.utils.ObservationArray`]

**Comment by rhiannonlynne** [comment]
@tribeiro said on slack that this is good for him - 
so looks good.

**Review by rhiannonlynne — APPROVED** [review]

**Pull request merged**